core/src/test/java/org/web3j/tx/ContractTest.java

```diff
@@ -168,45 +164,56 @@ public void testIsValidEmptyCode() throws Exception {
         assertFalse(contract.isValid());
     }
 
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     public void testIsValidNoBinThrows() throws Exception {
-        TransactionManager txManager = mock(TransactionManager.class);
-        TestContract contract =
-                new TestContract(
-                        Contract.BIN_NOT_PROVIDED,
-                        ADDRESS,
-                        web3j,
-                        txManager,
-                        new DefaultGasProvider());
-        contract.isValid();
+
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> {
+                    TransactionManager txManager = mock(TransactionManager.class);
+                    TestContract contract =
+                            new TestContract(
+                                    Contract.BIN_NOT_PROVIDED,
+                                    ADDRESS,
+                                    web3j,
+                                    txManager,
+                                    new DefaultGasProvider());
+                    contract.isValid();
+                });
     }
 
-    @Test(expected = RuntimeException.class)
+    @Test
     public void testDeployInvalidContractAddress() throws Throwable {
-        TransactionReceipt transactionReceipt = new TransactionReceipt();
-        transactionReceipt.setTransactionHash(TRANSACTION_HASH);
 
-        prepareTransaction(transactionReceipt);
-
-        String encodedConstructor =
-                FunctionEncoder.encodeConstructor(Arrays.<Type>asList(new Uint256(BigInteger.TEN)));
-
-        try {
-            TestContract.deployRemoteCall(
-                            TestContract.class,
-                            web3j,
-                            SampleKeys.CREDENTIALS,
-                            ManagedTransaction.GAS_PRICE,
-                            Contract.GAS_LIMIT,
-                            "0xcafed00d",
-                            encodedConstructor,
-                            BigInteger.ZERO)
-                    .send();
-        } catch (InterruptedException e) {
-            throw e;
-        } catch (ExecutionException e) {
-            throw e.getCause();
-        }
+        assertThrows(
+                RuntimeException.class,
+                () -> {
+                    TransactionReceipt transactionReceipt = new TransactionReceipt();
+                    transactionReceipt.setTransactionHash(TRANSACTION_HASH);
+
+                    prepareTransaction(transactionReceipt);
+
+                    String encodedConstructor =
+                            FunctionEncoder.encodeConstructor(
+                                    Arrays.<Type>asList(new Uint256(BigInteger.TEN)));
```

Comment:
Not sure if it's considered good practice to wrap entire function in `assertThrows`, or just the line that we actually expect to have issues with?

Reply:
Fixed in the latest commit.